### PR TITLE
chore: add E2E tests for torch training with nans

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
add a E2E test for torch training where the training data contains nans

### CODE REVIEW

File: python/python/lance/torch/kmeans.py [hunk 1]
```diff
@@ -249,6 +249,10 @@ def _fit_once(
             del dists
             del chunk
 
+        if total_dist == 0:
+            logging.warning("Kmeans::train: total_dist is 0, this is unusual")
```
Comment (eddyxu) [inline python/python/lance/torch/kmeans.py:253]: In what condition can we get here?
Can you add comments for explanation?

cc @QianZhu @albertlockett

File: python/python/lance/util.py [hunk 2]
```diff
@@ -176,3 +176,54 @@ def predict(
...
+    sample_size: int, optional
+        The number of vectors to sample from the dataset.
+        If None, the entire dataset will be used.
+    pass_threshold: float, default=1.0
+        The minimum fraction of vectors that must pass the sanity check.
+        If less than this fraction of vectors pass, a ValueError will be raised.
+    """
+
+    data = dataset.to_table() if sample_size is None else dataset.sample(sample_size)
+    vecs = data[column].to_numpy(zero_copy_only=False)
```
Comment (eddyxu) [inline python/python/lance/util.py:207]: can this handle `FixedShapedTensor`?

File: python/python/lance/util.py [hunk 3]
```diff
@@ -176,3 +176,54 @@ def predict(
...
+    passes = 0
+    total = len(vecs)
+
+    for vec in vecs:
+        if np.isnan(vec).any():
```
Comment (eddyxu) [inline python/python/lance/util.py:212]: this is like ignore nan query vector? what if the dataset are majority nans

File: python/python/lance/util.py (hunk 2 above)
Comment (chebbyChefNEQ) [reply python/python/lance/util.py:207]: I think so? I'll add a test

File: python/python/lance/util.py (hunk 3 above)
Comment (chebbyChefNEQ) [reply python/python/lance/util.py:212]: hmmm, good point let me do `total -= 0` instead

File: python/python/lance/util.py (hunk 2 above)
Comment (chebbyChefNEQ) [reply python/python/lance/util.py:207]: yeah works. Tensor is just FSL under the hood anyways

**Review by albertlockett — APPROVED** [review]

**Pull request merged**